### DISCUSSION TITLE
In .Net 5 Error : Query returned no results

### DISCUSSION THREAD

**Discussion by TechAspirant** [serenity-is/Serenity#5764] [Help]
I want to check whether name is unique or not. In order to validate on create and update I used below piece of code which checks well but while inserting unique name it throws error 'Query returned no results'

```
protected override void ValidateRequest()
            {
                base.ValidateRequest();
                if (IsCreate)
                {
                var f = MyRow.Fields;
                var d = Connection.First<MyRow>(f.AssetGroupName
                 == Row.AssetGroupName);
                if (d != null) throw new ValidationError("InvalidMyRow", "UniqueDepartamentIdAndJobId",
                                  "My error message");

                }
            }
```

**Comment by VictorTomaili** [comment]
Add a unique constraint to your database, this type of check has a micro transaction issue, if 2 requests at the same time executed, both can be inserted. Databases have features for this data consistency.

**Reply by TechAspirant** [reply]
Can I show custom error message when unique constraint throws error from SQL ?

**Reply by VictorTomaili** [reply]
You can create an ImplicitBehavior for customizing exceptions.

**Reply by TechAspirant** [reply]
Any example for ImplicitBehaviour or any wiki entry ?

**Reply by VictorTomaili** [reply]
You can check [NotesBehavior](https://github.com/serenity-is/Serene/blob/fe41ee63905f51b7ac1333ce0c3cce118ae5372f/Serene/Serene.Core/Modules/Northwind/Note/NotesBehavior.cs) in serene project. Also, you need to extend your behavior from `BaseSaveDeleteBehavior` to override `OnException` methods.

**Reply by TechAspirant** [reply]
Ok Thanks.
